### DISCUSSION TITLE
Ongoing data and formulas?

### DISCUSSION THREAD

**Discussion by jmgirard** [PittMethods/appliedml#23] [General]
I think it would probably be best to settle on around 2 to 4 datasets that we will use in this course. Then we can carry over the same examples and even syntax from lecture to lecture. Ideally they would be large enough to support cross-validation, include both continuous and categorical (probably just binary) labels, and be related to social/behavioral sciences. Perhaps we have one dataset for regression and one dataset for classification during lectures/live coding and then another dataset that supports both for practice activities.

**Comment by jmgirard** [comment]
For instance, whatever data and formulas you use in 1C and 1D will be what I use and expand into 2A and 2B.

**Comment by ShirleyBWang** [comment]
Yeah, I like that idea! I had been using `titanic` for both regression and classification (predicting survival + fare) and `affect` for data splitting but I forgot about `hcp_memory` which worked well last year for prediction tasks. Maybe we can stick with `titanic` and pick a few others? What do you prefer?

**Reply by jmgirard** [reply]
I think real (as opposed to simulated or modified) data should be prioritized if possible. Ideally, the variables would also be intuitively understandable. For these reasons, I don't like `hcp_memory` as much. I think `titanic` and `airsat` are good for classification (predicting survived and satisfaction, respectively). For regression, titanic fare is ok but not as interesting. I wonder if we could do better.

**Reply by jmgirard** [reply]
`water` is also interesting from a medicine and chemistry perspective, but is also classification (predicting potability).

**Reply by jmgirard** [reply]
Here is one where you use regression to predict medical costs:
https://www.kaggle.com/datasets/mirichoi0218/insurance

**Reply by ShirleyBWang** [reply]
I'm happy to give the medical costs one a try! I'm unfamiliar with that dataset and working on edits to 1C currently - if you'd like to download and take a look and decide you like it, then we can definitely go with that one.

**Reply by jmgirard** [reply]
I think for the early ones, having a single dataset for both would be ideal. So maybe go with `titanic` and we can move to a separate regression dataset later in the course if we find a good one.

**Reply by ShirleyBWang** [reply]
Sounds great!

**Reply by jmgirard** [reply]
This looks cool: 
https://www.kaggle.com/datasets/yasserh/song-popularity-dataset

**Reply by ShirleyBWang** [reply]
Oh that does look super cool! I'd be down if you wanted to try it

**Reply by jmgirard** [reply]
It appears to be data from Spotify and it is really huge (18000+ observations). Might be worth randomly subsetting it to make it more manageable for attendees to run. Maybe subset 8000 or so?

**Comment by jmgirard** [comment]
For reference, this is what we said in the advertisement:
> Although attendees of all backgrounds are welcome and the skills taught will be broadly applicable, example datasets and advice will be tailored specifically to the social and behavioral sciences (e.g., psychology, medicine, education, and related fields).

**Reply by ShirleyBWang** [reply]
Yeah, that's why we tried so hard to create/curate datasets last year, even if that meant simulating data. It ended up being OK but I wonder if things would be more straightforward with these simpler, widely-used datasets.
